core/src/test/java/org/jboss/jandex/test/AnnotationOverlayTest.java

```diff
@@ -233,97 +233,106 @@ private void assertOverlay(String expectedValues, AnnotationTransformation... tr
 
         for (boolean inheritedAnnotations : Arrays.asList(true, false)) {
             for (boolean runtimeAnnotationsOnly : Arrays.asList(true, false)) {
-                AnnotationOverlay.Builder builder = AnnotationOverlay.builder(index, Arrays.asList(transformations));
-                if (inheritedAnnotations) {
-                    builder.inheritedAnnotations();
-                }
-                if (runtimeAnnotationsOnly) {
-                    builder.runtimeAnnotationsOnly();
-                }
-                AnnotationOverlay overlay = builder.build();
-
-                StringBuilder values = new StringBuilder();
-
-                ClassInfo clazz = index.getClassByName(AnnotatedClass.class);
-                assertNotNull(clazz);
-
-                assertFalse(overlay.hasAnnotation(clazz, MyNotInheritedAnnotation.class));
-                assertNull(overlay.annotation(clazz, MyNotInheritedAnnotation.class));
-                assertEquals(0, overlay.annotationsWithRepeatable(clazz, MyNotInheritedAnnotation.class).size());
-
-                if (inheritedAnnotations) {
-                    assertTrue(overlay.hasAnnotation(clazz, MyInheritedAnnotation.class));
-                    assertNotNull(overlay.annotation(clazz, MyInheritedAnnotation.class));
-                    assertNotNull(overlay.annotation(clazz, MyInheritedAnnotation.class).target());
-                    assertEquals("i", overlay.annotation(clazz, MyInheritedAnnotation.class).value().asString());
-                    assertEquals(1, overlay.annotationsWithRepeatable(clazz, MyInheritedAnnotation.class).size());
-                } else {
-                    assertFalse(overlay.hasAnnotation(clazz, MyInheritedAnnotation.class));
-                    assertNull(overlay.annotation(clazz, MyInheritedAnnotation.class));
-                    assertEquals(0, overlay.annotationsWithRepeatable(clazz, MyInheritedAnnotation.class).size());
-                }
-
-                FieldInfo field = clazz.field("field");
-                assertNotNull(field);
+                for (boolean cacheAll : Arrays.asList(true, false)) {
```

Comment:
No idea how to test my changes properly, any ideas?
For now just expand the assertions, to make sure nothing breaks with cacheAll activated